File: back/DevC/src/main/java/com/back/devc/domain/post/post/dto/PostCreateResponse.java
package com.back.devc.domain.post.post.dto;

import com.back.devc.domain.post.post.entity.Post;

public record PostCreateResponse(
        Long postId
) {
    public static PostCreateResponse from(Post post) {
        return new PostCreateResponse(post.getPostId());
    }
}

File: back/DevC/src/main/kotlin/com/back/devc/domain/post/post/dto/PostCreateResponse.kt
package com.back.devc.domain.post.post.dto

import com.back.devc.domain.post.post.entity.Post

data class PostCreateResponse(
    val postId: Long
) {
    companion object {
        @JvmStatic
        fun from(post: Post): PostCreateResponse {
            return PostCreateResponse(
                postId = post.postId
                    ?: throw IllegalStateException("Post ID cannot be null"),
            )
        }
    }
}